src/insights_client/tests/test_motd.py

```diff
@@ -0,0 +1,114 @@
+import os
+import pathlib
+import tempfile
+import unittest
+import unittest.mock
+
+import pytest
+
+import insights_client
+
+
+class MockFS:
+    """Mock filesystem.
+
+    Ensures that the files required for the MOTD manipulation are created
+    in a temporary directory instead of working with actual system files.
+    """
+
+    _dir: tempfile.TemporaryDirectory
+
+    def __init__(self):
+        self._dir = tempfile.TemporaryDirectory(prefix="test-chroot-")
+
+        paths: dict[str, str] = {
+            "insights_client.MOTD_SRC": "/etc/insights-client/insights-client.motd",
+            "insights_client.MOTD_FILE": "/etc/motd.d/insights-client",
+            "insights_client.REGISTERED_FILE": "/etc/insights-client/.registered",
+            "insights_client.UNREGISTERED_FILE": "/etc/insights-client/.unregistered",
+        }
+
+        self.mocks: dict[str, unittest.mock.patch] = {}
+        for label, path in paths.items():
```

Comment:
I was a bit confused by the variable name _label_. It gave me an impression that it’s some arbitrary naming string, but it fact, it’s the constant path and name to be patched. [unittest.mock](https://docs.python.org/3.6/library/unittest.mock.html) [calls](https://docs.python.org/3.6/library/unittest.mock.html#unittest.mock.patch) it “target”.

```suggestion
        for target, path in paths.items():
```